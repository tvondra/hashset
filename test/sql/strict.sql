/*
 * Test to verify all relevant functions return NULL if any of
 * the input parameters are NULL, i.e. testing that they are declared as STRICT
 */

SELECT hashset_add(int4hashset(), NULL::int);
SELECT hashset_add(NULL::int4hashset, 123::int);
SELECT hashset_contains('{123,456}'::int4hashset, NULL::int);
SELECT hashset_contains(NULL::int4hashset, 456::int);
SELECT hashset_merge('{1,2}'::int4hashset, NULL::int4hashset);
SELECT hashset_merge(NULL::int4hashset, '{2,3}'::int4hashset);
SELECT hashset_to_array(NULL::int4hashset);
SELECT hashset_count(NULL::int4hashset);
SELECT hashset_capacity(NULL::int4hashset);
SELECT hashset_intersection('{1,2}'::int4hashset,NULL::int4hashset);
SELECT hashset_intersection(NULL::int4hashset,'{2,3}'::int4hashset);
SELECT hashset_difference('{1,2}'::int4hashset,NULL::int4hashset);
SELECT hashset_difference(NULL::int4hashset,'{2,3}'::int4hashset);
SELECT hashset_symmetric_difference('{1,2}'::int4hashset,NULL::int4hashset);
SELECT hashset_symmetric_difference(NULL::int4hashset,'{2,3}'::int4hashset);

/*
 * For convenience, hashset_agg() is not STRICT and just ignore NULL values
 */
SELECT hashset_agg(i) FROM (VALUES (NULL::int),(1::int),(2::int)) q(i);

SELECT hashset_agg(h) FROM
(
    SELECT NULL::int4hashset AS h
    UNION ALL
    SELECT hashset_agg(j) AS h FROM generate_series(6,10) AS j
) q;
