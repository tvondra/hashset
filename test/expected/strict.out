/*
 * Test to verify all relevant functions return NULL if any of
 * the input parameters are NULL, i.e. testing that they are declared as STRICT
 */
SELECT hashset_add(int4hashset(), NULL::int);
 hashset_add 
-------------
 
(1 row)

SELECT hashset_add(NULL::int4hashset, 123::int);
 hashset_add 
-------------
 
(1 row)

SELECT hashset_contains('{123,456}'::int4hashset, NULL::int);
 hashset_contains 
------------------
 
(1 row)

SELECT hashset_contains(NULL::int4hashset, 456::int);
 hashset_contains 
------------------
 
(1 row)

SELECT hashset_merge('{1,2}'::int4hashset, NULL::int4hashset);
 hashset_merge 
---------------
 
(1 row)

SELECT hashset_merge(NULL::int4hashset, '{2,3}'::int4hashset);
 hashset_merge 
---------------
 
(1 row)

SELECT hashset_to_array(NULL::int4hashset);
 hashset_to_array 
------------------
 
(1 row)

SELECT hashset_count(NULL::int4hashset);
 hashset_count 
---------------
              
(1 row)

SELECT hashset_capacity(NULL::int4hashset);
 hashset_capacity 
------------------
                 
(1 row)

SELECT hashset_intersection('{1,2}'::int4hashset,NULL::int4hashset);
 hashset_intersection 
----------------------
 
(1 row)

SELECT hashset_intersection(NULL::int4hashset,'{2,3}'::int4hashset);
 hashset_intersection 
----------------------
 
(1 row)

SELECT hashset_difference('{1,2}'::int4hashset,NULL::int4hashset);
 hashset_difference 
--------------------
 
(1 row)

SELECT hashset_difference(NULL::int4hashset,'{2,3}'::int4hashset);
 hashset_difference 
--------------------
 
(1 row)

SELECT hashset_symmetric_difference('{1,2}'::int4hashset,NULL::int4hashset);
 hashset_symmetric_difference 
------------------------------
 
(1 row)

SELECT hashset_symmetric_difference(NULL::int4hashset,'{2,3}'::int4hashset);
 hashset_symmetric_difference 
------------------------------
 
(1 row)

/*
 * For convenience, hashset_agg() is not STRICT and just ignore NULL values
 */
SELECT hashset_agg(i) FROM (VALUES (NULL::int),(1::int),(2::int)) q(i);
 hashset_agg 
-------------
 {1,2}
(1 row)

SELECT hashset_agg(h) FROM
(
    SELECT NULL::int4hashset AS h
    UNION ALL
    SELECT hashset_agg(j) AS h FROM generate_series(6,10) AS j
) q;
 hashset_agg  
--------------
 {6,7,8,9,10}
(1 row)

